debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresSnapshotChangeEventSource.java

```diff
@@ -211,9 +212,40 @@ protected void complete(SnapshotContext<PostgresOffsetContext> snapshotContext)
         snapshotter.snapshotCompleted();
     }
 
+    @Override
+    protected String enhanceOverriddenSelect(RelationalSnapshotContext<PostgresOffsetContext> snapshotContext, String overriddenSelect, TableId tableId) {
+        String snapshotSelectColumns = getSnapshotSelectColumns(tableId);
+        return overriddenSelect.replaceAll(SELECT_ALL_PATTERN.pattern(), snapshotSelectColumns);
+    }
+
     @Override
     protected Optional<String> getSnapshotSelect(RelationalSnapshotContext<PostgresOffsetContext> snapshotContext, TableId tableId) {
```

Comment:
So in terms of the overall contract, I think this is was we should have:

```
protected Optional<String> getSnapshotSelect(RelationalSnapshotContext<PostgresOffsetContext> snapshotContext, TableId tableId, List<String> columns);

protected String resolveColumnName(TableId tableId, String columnName);
```

The base class then would do the filtering, invoke `resolveColumnName()` for each included column, and invoke `getSnapshotSelect()` with all the resolved columns. This should allow for the least amount of duplication, while providing the connector-specific sub-classes the flexibility they need.